SU2_CFD/include/driver_structure.hpp

```diff
@@ -1481,3 +1528,175 @@ class CMultiphysicsZonalDriver : public CDriver {
   void Transfer_Data(unsigned short donorZone, unsigned short targetZone);
 
 };
+
+/*!
+ * \class CSinglezoneDriver
+ * \brief Class for driving single-zone solvers.
+ * \author R. Sanchez
+ * \version 6.0.1 "Falcon"
+ */
+class CSinglezoneDriver : public CDriver {
+protected:
+
+  unsigned long TimeIter;
+
+public:
+
+  /*!
+   * \brief Constructor of the class.
+   * \param[in] confFile - Configuration file name.
+   * \param[in] val_nZone - Total number of zones.
+   * \param[in] MPICommunicator - MPI communicator for SU2.
+   */
+  CSinglezoneDriver(char* confFile,
+             unsigned short val_nZone,
+             unsigned short val_nDim,
+             bool val_periodic,
+             SU2_Comm MPICommunicator);
+
+  /*!
+   * \brief Destructor of the class.
+   */
+  ~CSinglezoneDriver(void);
+
+  /*!
+   * \brief [Overload] Launch the computation for single-zone problems.
+   */
+  void StartSolver();
+
+  /*!
+   * \brief Preprocess the single-zone iteration
+   */
+  void Preprocess(unsigned long TimeIter);
+
+  /*!
+   * \brief Run the iteration for ZONE_0.
+   */
+  void Run();
+
+  /*!
+   * \brief Use a corrector step to prevent convergence issues.
+   */
+  void Corrector();
```

Comment:
It depends a lot on the discipline. Some people talk about predictor-corrector strategies and that's where I took the name from. It could also be named relaxation (but we can always change the name later on).

Reply:
Okay, understood.